shared/AppInsightsCore/src/JavaScriptSDK/HelperFuncs.ts

```diff
@@ -781,9 +781,18 @@ export function proxyAssign<T, S>(target: T, source: S, chkSet?: (name: string,
     return target;
 }
 
-export function proxyFunctionAs<T, S>(target: T, name: string, source: S | (() => S), theFunc: (keyof S), overwriteTarget: boolean = true) {
+/**
+ * Creates a proxy function on the target which internally will call the source version with all arguments passed to the target method.
+ *
+ * @param target - The target object to be assigned with the source properties and functions
+ * @param name - The function name that will be added on the target
+ * @param source - The source object which will be assigned / called by setting / calling the targets proxies
+ * @param theFunc - The function name on the source that will be proxied on the target
+ * @param overwriteTarget - If `false` this will not replace any pre-existing name otherwise (the default) it will overwrite any existing name
+ */
+export function proxyFunctionAs<T, S>(target: T, name: string, source: S | (() => S), theFunc: (keyof S), overwriteTarget?: boolean) {
     if (target && name && source) {
-        if (overwriteTarget || isUndefined(target[name])) {
+        if (overwriteTarget !== false || isUndefined(target[name])) {
```

Comment:
When overwriteTarget = undefined will return true in fist condition, is that handled?

Reply:
yes as `undefined !== false` which is what I want / need as the default because TypeScript will define the stub prototype methods during unit testing. So not overwriting fails the test.